packages/client/services/filter_service.ts

```diff
@@ -70,5 +71,12 @@ export function filterTransctions(
         }
     });
 
+    if (contactFilters.length !== 0) {
+        transactions = transactions.filter(
+            (transaction) =>
+                contactFilters.includes(transaction.contactIn?.contactName ?? "") ||
+                contactFilters.includes(transaction.contactOut?.contactName ?? ""),
```

Comment:
The logic right now will include transactions that aren't associated with a contact which isn't right. If contact name is null then `contactFilters.includes("")` will be true always. Just check before the includes if one is present, if not then shortcircuit

```suggestion
                transaction.contactIn?.contactName && contactFilters.includes(transaction.contactIn.contactName) ||
                transaction.contactOut?.contactName && contactFilters.includes(transaction.contactOut.contactName),
```

Reply:
that was kinda the idea. I wanted to include transactions without a contact. But looking back it does not make sense